model.py

```diff
@@ -257,17 +234,22 @@ def _run(self, inputs, labels=None, device='CPU', device_ids=None):
         inputs = to_list(inputs)
         if labels is not None:
             labels = to_list(labels)
-        assert len(inputs) == len(self._input_desc), "number of inputs" \
+        assert len(inputs) == len(self.model._inputs), "number of inputs" \
             + " does not match number of arguments of `forward` method"
 
         if self._progs.get(self.mode, None) is None:
-            self._make_program(self._infer_input_vars(inputs))
+            if self.model._inputs is None:
+                raise ValueError("The inputs of Model must be not None.")
+            self._input_vars = [
+                k.forward() for k in to_list(self.model._inputs)
+            ]
+            self._make_program(self._input_vars)
```

Comment:
isn't this supposed to happen in `prepare` now?

Reply:
This PR is only for loss, not change other module.  If this PR is accepted, can do this.

Reply:
Move to prepare